homework-assignments-1/task-2.js

```diff
@@ -0,0 +1,22 @@
+class User {
+    #password = "helloworld123";
+
+    constructor(){
+       this._username = "John Doe";
+    }
+
+
+    validateLogin(inputPassword) {
```

Comment:
The login validation process should be handled internally by the method without exposing the details of how passwords are compared.

`  public validateLogin(inputPassword: string): boolean {
    // Abstracts the logic of password comparison
    return this.checkPassword(inputPassword);
  }

  // Private method for password comparison
  private checkPassword(inputPassword: string): boolean {
    // Compare the input password with the actual password
    return inputPassword === this.password;
  }`